### PR TITLE
DOCSP-45314: Removed headers

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-45314
Staging - https://deploy-preview-84--docs-cpp.netlify.app/upgrade/#version-4.0

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [ ] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/upgrade.txt [hunk 1]
```diff
@@ -159,6 +159,13 @@ The v4.0 driver introduces the following breaking changes to the API:
   library equivalents. For context, see the :ref:`Build System Breaking Changes
   <version-4.0-build-system-breaking>` section.
 
+- Removes the following headers:
```
Comment (rustagir) [inline source/upgrade.txt:162]: Q: would C++ devs be familiar with the term "headers" or do you think some more explanation could be useful here?

**Review by rustagir — APPROVED** [review]
left a question but lgtm

File: source/upgrade.txt (hunk 1 above)
Comment (norareidy) [reply source/upgrade.txt:162]: I changed to "header files" to be more specific, but both headers/header files are mentioned throughout the docs

**Review by eramongodb — APPROVED** [review]
One suggestion; otherwise, LGTM.

**Review by eramongodb — COMMENTED** [review]
Whoops, missed the feedback.

**Pull request merged**